post_office/utils.py

```diff
@@ -138,14 +138,22 @@ def parse_emails(emails):
     return emails
 
 
-def cleanup_expired_mails(cutoff_date, delete_attachments=True):
+def cleanup_expired_mails(cutoff_date, delete_attachments=True, batch_size=10000):
     """
     Delete all emails before the given cutoff date.
     Optionally also delete pending attachments.
     Return the number of deleted emails and attachments.
     """
-    expired_emails = Email.objects.only('id').filter(created__lt=cutoff_date)
-    emails_count, _ = expired_emails.delete()
+    expired_emails = Email.objects.only('id').filter(created__lt=cutoff_date).values_list('id', flat=True)
+    total_deleted_emails = 0
+    while expired_emails:
```

Comment:
this will evaluate the queryset to hit DB to check expired email is existing every time loop continue, and on line 150 will hit DB again to limit it based on batch_size

what i mean is before the loop we need to fetch expired email ids first
then batch the ids to delete then delete it
```
expired_email_ids = list(Email.objects.only('id').filter(created__lt=cutoff_date).values_list('id', flat=True))
total_expired_emails = len(expired_email_ids)
offset = 0
email_ids = expired_email_ids[offset:batch_size]
while email_ids:
   deleted_data = Email.objects.filter(id__in=email__ids).delete()
   offset += deleted_data['post_office.Email']
   email_ids = expired_email_ids[offset:offset + batch_size]
```